### PR TITLE
Move `base/validate` to Validation

### PR DESCRIPTION
**Pull request by armiol**
This changeset moves `base/validate` module to this repository under `java-runtime-bundle` name.

Obsolete `runtime` and `runtime-extensions` modules are removed. 

Also, all dependencies to `mc-java` now use its fat JAR (partially addressing [mc-java#53](https://github.com/SpineEventEngine/mc-java/issues/53)).

Other changes include the updates to `config` contents: new Gradle version, updated Kotlin, etc.

### CODE REVIEW

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

File: java-runtime-bundle/src/main/java/io/spine/validate/ExternalConstraintOptions.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+     * @param <T>
+     *         the type of the option value
+     * @return the {@code Optional} of option value or {@code Optional.empty()}
+     *         if there is no such an option for the field descriptor
+     */
+    public static <T> Optional<T>
+    getOptionValue(FieldContext fieldContext, GeneratedExtension<FieldOptions, T> option) {
+        var options = Holder.instance.options;
+        for (var context : options.keySet()) {
+            var fieldOptions = options.get(context);
```
[the remark this answers is not in the document]
Comment (armiol) [reply java-runtime-bundle/src/main/java/io/spine/validate/ExternalConstraintOptions.java]: The key value is also used. Therefore, this is a false-positive.

**Comment by alexander-yevsyukov** [comment]
@armiol, why do we need the `bundle` suffix in this case?

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with the suggestion to shorten artifact name, removing the `bundle` suffix.

**Comment by armiol** [comment]
@alexander-yevsyukov we need this suffix to distinguish this artifact from the one with the same name we used previously. "Old" `java-runtime` was a completely different beast. And it has its versions published as well. I don't want new artifact to make an impression that it is the same thing as the "old" one. 

Besides, with this suffix it is totally clear it is a fat JAR assembly. I could use "fat" instead, but it seems a bit of a jargon. "Bundle" looks more official to me.

**Comment by alexander-yevsyukov** [comment]
@armiol, got it. Agreed.

**Pull request merged**